core/testing/router.go

```diff
@@ -0,0 +1,159 @@
+package coretesting
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+
+	appmodulev2 "cosmossdk.io/core/appmodule/v2"
+	"cosmossdk.io/core/router"
+	"cosmossdk.io/core/transaction"
+)
+
+var ErrNoHandler = errors.New("no handler")
+
+// NewMsgRouterBuilder is a router that routes messages to their respective handlers.
+func NewMsgRouterBuilder() *ReflectionRouterBuilder {
```

Comment:
my IDE says there are no usages of this function - is there an example usage of this somewhere?

Reply:
This would be used for an `appmodulev2` module - which this task doesn't touch

Reply:
adding comment here from our conversation so we don't forget: resolution here is to add a test that utilizes this code in some way